web/middleware.go

```diff
@@ -128,3 +130,30 @@ func forbidAccess(c *gin.Context, cav string) {
 	c.JSON(http.StatusForbidden, gin.H{"reason": fmt.Sprintf("caveat \"%s\" not satisfied", cav)})
 	c.Abort()
 }
+
+func checkParticipant(filename string) gin.HandlerFunc {
+	var participants = map[string]struct{}{}
+
+	if filename == "" {
+		panic("invalid participant file given")
+	}
+
+	if f, err := os.Open(filename); err != nil {
```

Comment:
The file is not closed.

The file reading can be handled outside the handler function. This function simply uses the map `participants` to check if the participant is valid.

Reply:
This `Open` will only run once. It is not the handler itself.

Reply:
@nodestory Fix the file closing issue